### PR TITLE
 Update TypeScript to 2.7.2 and clean up compilation errors

### PR DESCRIPTION
**Pull request by davidsoergel**
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/113)

### CODE REVIEW

File: src/types_test.ts [hunk 1]
```diff
@@ -94,10 +94,6 @@ describeMathCPU('ConcreteTensor Test', () => {
   it('Read value', () => {
     const v1 = new CT(scalar(10), 'foo');
     expect(v1.value().dataSync()).toEqual(new Float32Array([10]));
-
-    let v1Value = v1.value();
```
Comment (davidsoergel) [inline src/types_test.ts]: I don't quite understand the original intent of these deleted lines.  Please let me know if there's something that should actually be tested here.

**Comment by caisq** [comment]
:lgtm_strong:

Thanks, David!

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/113#-:-L9GwjA92Njyhx41lUTl:b-geg47d)*

**Comment by nsthorat** [comment]
:lgtm_strong:

WOohoooo thank you for doing this! My VSCode was using 2.7 and the build was using 2.4 so I was getting errors in my editor but not when I ran it. This is great :)

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/113#-:-L9IQ8OE1q2a35oowC8a:bs9e8cn)*

**Comment by nsthorat** [comment]
FYI your travis is failing because of lint complaining about cyclic deps: https://github.com/tensorflow/tfjs-layers/blob/master/tslint.json#L2

I would just turn it off. We turned off the cyclic dep check in core (actually, cyclic dependencies are fine as long as you dont refer to symbols during module initialization time, and this linter rule is actually really really slow).

**Comment by caisq** [comment]
David is looking into the circular dependencies, I believe. Some of them look like false positives; others look genuine.

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/113#-:-L9IRr4--jPV12XbOcnN:b-924ens)*

**Comment by nsthorat** [comment]
Awesome, just FYI it's completely fine to have circular dependencies in the JavaScript world as long as there are no module-loading time references to other modules (basically code that runs immediately and references another module when the file is loaded, outside of a function body).

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/113#-:-L9ISJvf9JGthw2nHWhZ:btgyiv0)*

**Comment by dsmilkov** [comment]
Yeah, I also recommend to disable tslint-circular-dep, it will increase the `yarn build` dramatically, and not all circular deps are bad design (e.g. we have chaining api in tfjs-core and didn't want to write a whole interface of Tensor {} just to avoid an artificial problem)

**Pull request merged**